src/shared-components/container/style.ts

```diff
@@ -0,0 +1,99 @@
+import { DetailedHTMLProps, HTMLAttributes } from 'react';
+import styled, { StyledComponent } from '@emotion/styled';
+import { css } from '@emotion/core';
+import PropTypes from 'prop-types';
+
+import {
+  SPACER, BOX_SHADOWS, COLORS, MEDIA_QUERIES, 
+} from '../../constants';
+
+const clickableStyle = css`
+  box-shadow: ${BOX_SHADOWS.clickable};
+  cursor: pointer;
+  transition: box-shadow 200ms;
+
+  &:hover {
+    box-shadow: ${BOX_SHADOWS.clickableHover};
+  }
+`;
+
+const messageStyle = css`
+  border-radius: 16px;
+  box-shadow: ${BOX_SHADOWS.message};
+`;
+
+export type ContainerType = 'clickable' | 'message';
+
+const getContainerTypeStyles = (type?: ContainerType) => {
+  switch (type) {
+    case 'message':
+      return messageStyle;
+    case 'clickable':
+      return clickableStyle;
+    default:
+      return css``;
+  }
+};
+
+export const containerStyles = (type?: ContainerType) => css`
+  background-color: ${COLORS.white};
+  border: 1px solid ${COLORS.border};
+
+  ${getContainerTypeStyles(type)};
+`;
+
+// TODO update desktop padding to 48px
+export const Section = styled.div`
+  padding: ${SPACER.large};
+
+  ${MEDIA_QUERIES.lgUp} {
+    padding-left: ${SPACER.x2large};
+    padding-right: ${SPACER.x2large};
+  }
+`;
+
+export const Divider = styled.div`
+  margin: 0 ${SPACER.large};
+  border-bottom: 1px solid ${COLORS.divider};
+
+  ${MEDIA_QUERIES.lgUp} {
+    margin: 0 ${SPACER.x2large};
+  }
+`;
+
+export const Image = styled.img`
+  width: 100%;
+  overflow: hidden;
+  object-fit: cover;
+`;
+
+type StyledContainer = StyledComponent<
+  DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
+  {
+    type?: 'clickable' | 'message' | undefined;
+  },
+  any
```

Comment:
[silly question] As a TS novice, I still find some of these type definitions to be difficult to parse. For example, what does the `any` refer to? Is that part of the `DetailedHTMLProps` that we expect? My best attempt at explaining this: 

We're defining a type called `StyledContainer`, which is a `StyledComponent` which accepts `DetailedHTMLProps` as an argument, plus the additional attribute of `type` optionally.

Reply:
I've added a code comment in 7926094 but this is just the type that `@emotion/styled` returns 😬 , so I don't think there's a way for people to get comfortable with advanced and complex types beyond working more directly/frequently with these libraries and browsing the type definitions. 

So `any` corresponds to the Theme: `StyledComponent<InnerProps, StyleProps, Theme>`